### DISCUSSION TITLE
Keep volume value to easily toggle mute

### DISCUSSION THREAD

**Discussion by clsty** [end-4/dots-hyprland#442] [Ideas]
Right now when we press "Mute" key (media key) it always goes mute.
Pressing it again does nothing, you need to press "Volume up" again to toggle back.

But what if we just want to temporarily mute for some seconds, and after that quickly resume the previous volume?

It should be relatively easy to implement with shell script or AGS.

If with shell script:
When press "Mute" it also write the current volume value to a temp file such as `/tmp/volume.value.txt`. And when "Mute" is pressed again, it detects that it is currently in mute state so it restores the previous volume value, and done.
Note that user should still be allowed to increase volume from zero, in such case the `/tmp/volume.value.txt` should sync to current value too.

This one is not very important but some people may like it, which I'm not sure about. Feel free to talk more about it.
Despite that I don't have much time for implementing it.

**Comment by end-4** [comment]
There's actually a normal way to (un)mute with that behavior. I was just lazy to implement the OSD popup so I had the key set volume to 0
